### DISCUSSION TITLE
kustomization is healthy but is not being applied to the expected resource

### DISCUSSION THREAD

**Discussion by emily-zall** [runwhen-contrib/runwhen-local#410] [Ideas]
My flux kustomization is healthy but is not being applied to the expected resource - https://github.com/fluxcd/kustomize-controller/issues/1047

some troubleshooting commands I tried
```
kubectl logs -n flux-system kustomize-controller-859c949c64-l848t | grep -i strimzi-kafka-operator
kubectl logs -n flux-system kustomize-controller-859c949c64-l848t | grep -i error
```
In my case this didn't give me a useful error message but the first of those commands did show that the kustomize controller was seeing my helmrelease as changed.

I don't know if there is any further troubleshooting commands that can be run to get a useful message.

**Comment by stewartshea** [comment]
Looking at the response in the issue above, I'm trying to come up with a few commands that would help out here; 

Maybe; 
- Check for deprecated apiVersions
- Check FluxCD versions

Without an error messsage, I'm not quite sure how we would really know that something else was wrong. I'd be very curious if you're able to resolve it, and if so, if you have other ideas of a command we could run to identify the issue.

**Reply by emily-zall** [reply]
Thanks! Here's the resolution: https://github.com/fluxcd/kustomize-controller/issues/1047 There was a deprecated apiVersion in the mix so that could be helpful though it wasn't the cause this time. It seems like if there are no matches to the target kustomize doesn't indicate that in anyway. But what does the flux kustomize controller log when it does find a match? I think some type of querires on the logs like `kubectl logs -n flux-system kustomize-controller-859c949c64-l848t | grep -i strimzi-kafka-operator | grep -v unchanged` could be helpful even if they weren't decisive here and if I knew what logs to expect in a successful case then knowing what was missing would provide helpful information.

**Reply by stewartshea** [reply]
Thanks for the idea, I'll play with this a little bit to see what task we can add that makes the most amount of sense. I'll try to reproduce and see what we can drum up.
